JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/home/HomeScreeViewModel.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetstream.presentation.screens.home
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.jetstream.data.entities.MovieList
+import com.google.jetstream.data.repositories.MovieRepository
+import com.google.jetstream.data.repositories.MovieRepositoryImpl
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+
+class HomeScreeViewModel(movieRepository: MovieRepository = MovieRepositoryImpl.instance): ViewModel() {
+
+    val featuredMovieList: StateFlow<MovieList> = movieRepository.getFeaturedMovies().stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5_000),
+        initialValue = MovieList.Empty
+    )
+
+    val trendingMovieList: StateFlow<MovieList> = movieRepository.getTrendingMovies().stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5_000),
+        initialValue = MovieList.Empty
+    )
+
+    val top10MovieList: StateFlow<MovieList> = movieRepository.getTop10Movies().stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5_000),
+        initialValue = MovieList.Empty
+    )
+
+    val nowPlayingMovieList: StateFlow<MovieList> = movieRepository.getNowPlayingMovies().stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5_000),
+        initialValue = MovieList.Empty
+    )
+
```

Comment:
Create a sealed interface representing this screen's state and expose single state property:

```
sealed interface HomeScreenUiState {
   object Loading: HomeScreenUiState
   object Error: HomeScreenUiState
   data class Content(
      val featuredMovieList: List<MovieList>,
      val trendingMovieList: List<MovieList>,
      val top10MovieList: : List<MovieList>,
      val nowPlayingMovieList: : List<MovieList>
   )
}

val uiState: StateFlow<HomeScreenUiState> = ... 
```